unison-core/src/Unison/Project.hs

```diff
@@ -104,11 +104,11 @@ instance From ProjectBranchName Text
 
 instance TryFrom Text ProjectBranchName where
   tryFrom =
-    maybeTryFrom (Megaparsec.parseMaybe projectBranchNameParser)
+    maybeTryFrom (Megaparsec.parseMaybe (projectBranchNameParser True))
```

Comment:
Some Boolean blindness here; wdyt about a new type for AllowLeadingSlash

Reply:
I think it's ok as-is, the boolean arguments are private to this module